Framework/Kernel/src/TimeROI.cpp

```diff
@@ -99,17 +54,57 @@ void TimeROI::addROI(const std::string &startTime, const std::string &stopTime)
  */
 void TimeROI::addROI(const Types::Core::DateAndTime &startTime, const Types::Core::DateAndTime &stopTime) {
   assert_increasing(startTime, stopTime);
-  if ((this->empty()) || (startTime > m_roi.lastTime()) || (stopTime < m_roi.firstTime())) {
+
+  if ((this->empty()) || (startTime > m_roi.back())) {
     // add in the new region
-    m_roi.addValue(startTime, ROI_USE);
-    m_roi.addValue(stopTime, ROI_IGNORE);
-  } else if (this->isCompletelyInROI(startTime, stopTime)) {
-    g_log.debug("TimeROI::addROI is already accounted for and being ignored");
+    m_roi.push_back(startTime);
+    m_roi.push_back(stopTime);
+  } else if (stopTime < m_roi.front()) {
+    m_roi.insert(m_roi.begin(), stopTime);
```

Comment:
I believe inserting elements in a vector is costly because of memory reallocation. Do you think we should worry about this?

Reply:
Unless you know how big the vector should be from the get-go, I'm not sure how to do this better. In general, you are right though.